pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -0,0 +1,79 @@
+package rest
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+	"k8s.io/klog"
+)
+
+type DualWriterMode2 struct {
+	DualWriter
+}
+
+// NewDualWriterMode2 returns a new DualWriter in mode 2.
+// Mode 2 represents writing to LegacyStorage and Storage and reading from LegacyStorage.
+func NewDualWriterMode2(legacy LegacyStorage, storage Storage) *DualWriterMode2 {
+	return &DualWriterMode2{*newDualWriter(legacy, storage)}
+}
+
+// Create overrides the default behavior of the DualWriter and writes to LegacyStorage and Storage.
+func (d *DualWriterMode2) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
+	legacy, ok := d.legacy.(rest.Creater)
+	if !ok {
+		return nil, fmt.Errorf("legacy storage rest.Creater is missing")
+	}
+
+	created, err := legacy.Create(ctx, obj, createValidation, options)
+	if err != nil {
+		return created, err
+	}
+
+	c, err := enrichObject(obj, created)
+	if err != nil {
+		return created, err
+	}
+
+	accessor, err := meta.Accessor(c)
+	if err != nil {
+		return created, err
+	}
+	accessor.SetResourceVersion("")
```

Comment:
any reason why we are setting the resource version to empty?

Reply:
Addressed this and other comments in the PoC PR: https://github.com/grafana/grafana/pull/84605